src/client/render2d/pixi_renderer.tsx

```diff
@@ -0,0 +1,119 @@
+import * as classNames from 'classnames'
+import { IReactionDisposer } from 'mobx'
+import { observable } from 'mobx'
+import { action } from 'mobx'
+import { autorun } from 'mobx'
+import { observer } from 'mobx-react'
+import { WebGLRenderer } from 'pixi.js'
+import { CanvasRenderer } from 'pixi.js'
+import { Container } from 'pixi.js'
+import { autoDetectRenderer } from 'pixi.js'
+import * as React from 'react'
+import { Component } from 'react'
+import ReactResizeDetector from 'react-resize-detector'
+
+import { Transform } from '../math/transform'
+
+import { renderObject2d } from './pixi/rendering'
+import { RendererProps } from './renderer_props'
+import * as style from './style.css'
+
+@observer
+export class PixiRenderer extends Component<RendererProps> {
+  @observable private resolution: Transform = Transform.of()
+  private canvas: HTMLCanvasElement
+  private renderer: WebGLRenderer | CanvasRenderer
+  private stopAutorun: IReactionDisposer
+
+  componentDidMount() {
+    if (!this.canvas) {
+      return
+    }
+
+    this.renderer = autoDetectRenderer({
+      view: this.canvas,
+      transparent: true,
+      antialias: true,
+    })
+
+    // Render when changes happen
+    this.stopAutorun = autorun(() => this.renderCanvas(), {
```

Comment:
unnecessary lambda here
```ts
this.stopAutorun = autorun(this.renderCanvas, {
...
})
```

Reply:
This has been done on every single autorun call throughout the codebase.
Should I change them all?

Reply:
Seems it's important for the localisation one at least, the other two work with this change